nemo/lightning/pytorch/strategies.py

```diff
@@ -697,18 +697,19 @@ def restore_checkpoint_after_setup(self) -> bool:
 
     @property
     def parallelism(self):
-        from megatron.core.model_parallel_config import ModelParallelConfig
-
-        return ModelParallelConfig(
-            tensor_model_parallel_size=self.tensor_model_parallel_size,
-            pipeline_model_parallel_size=self.pipeline_model_parallel_size,
-            virtual_pipeline_model_parallel_size=self.virtual_pipeline_model_parallel_size,
-            context_parallel_size=self.context_parallel_size,
-            sequence_parallel=self.sequence_parallel,
-            expert_model_parallel_size=self.expert_model_parallel_size,
-            moe_extended_tp=self.moe_extended_tp,
-            pipeline_dtype=self.pipeline_dtype,
-        )
+        from collections import namedtuple
+
+        kwargs = {}
+        kwargs['tensor_model_parallel_size'] = self.tensor_model_parallel_size
+        kwargs['pipeline_model_parallel_size'] = self.pipeline_model_parallel_size
+        kwargs['virtual_pipeline_model_parallel_size'] = self.virtual_pipeline_model_parallel_size
+        kwargs['context_parallel_size'] = self.context_parallel_size
+        kwargs['sequence_parallel'] = self.sequence_parallel
+        kwargs['expert_model_parallel_size'] = self.expert_model_parallel_size
+        kwargs['moe_extended_tp'] = self.moe_extended_tp
+        kwargs['pipeline_dtype'] = self.pipeline_dtype
+        ParallelismConfig = namedtuple('ParallelismConfig', ', '.join(kwargs.keys()))
```

Comment:
I think it would be better to create a dataclass for this outside of this property and construct it inside. That way this property can also have a output type-hint.

Reply:
Thanks, I've added a dataclass for this. PTAL.